### PR TITLE
[FIX] website_sale: discount

### LINKED ISSUE #31337: Pricelist: "Show public price & discount to the customer" not working in order created from website
### Impacted versions:
11.0

### Steps to reproduce:
1. "Sales / Configuration / Settings", search by "Multiple Sales Prices per Product" and select option "Prices computed from formulas (discounts, margins, roundings)"
2. Go to "Website / Catalog / Pricelists", select "Public Pricelist" and set "Discount Policy: Show public price & discount to the customer"
3. In another tab, please go to the Shop in the website and search by product "iPad Retina Display" (this product has a discount)
4. Add product to cart and then click in button "Proceed to Checkout"
5. In the backend, please go to "Website / Orders / Orders "
6. Remove the default search filter "Confirmed Orders" and search by Customer "Administrator"

### Current behavior:

When I check the create sale order in the backend it shows the "Ipad Retina Display" line with the discount applied directly to the unit price. it does not show the information related to the corresponding discounts.

image

### Expected behavior:

If in the same sale from the backend (or in any other) I add the same product, you can see that the line have the detailed and real information about the product price and the applied discount. 

image

image

-- tested in http://456445-11-0-5917ef.runbot15.odoo.com/

### PR DESCRIPTION
**Pull request by nim-odoo**
- Go to Sales > Configuration > Settings
- Enable "Multiple Sales Prices per Product" and select option "Prices
  computed from formulas (discounts, margins, roundings)"
- Go to Sales > Catalog / Pricelists
- Select "Public Pricelist" and set "Discount Policy: Show public price
  & discount to the customer"
- Go to the eCommerce, buy "iPad Retina Display" which as a 20 %
  discount

In the corresponding SO, the discount appears to be zero, and the
discount is included in the price unit. However, adding the product
directly to the SO in the backend displays the discount as expected.

There should not be a difference in behavior between the eCommerce and
the backend.

Fixes #31337
opw-1943989

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by nim-odoo** [comment]
@JKE-be I'm not sure we should really do this. From my experience, we already had quite a lot of issues with discount and pricelists on the website, and I'm afraid it could cause more problems than it solves.

File: addons/website_sale/models/sale_order.py [hunk 1]
```diff
@@ -98,18 +98,34 @@ def _website_product_id_change(self, order_id, product_id, qty=0):
...
         product = self.env['product.product'].with_context(product_context).browse(product_id)
-        pu = product.price
-        if order.pricelist_id and order.partner_id:
-            order_line = order._cart_find_product_line(product.id)
-            if order_line:
-                pu = self.env['account.tax']._fix_tax_included_price_company(pu, product.taxes_id, order_line[0].tax_id, self.company_id)
+        discount = 0
+
+        if order.pricelist_id.discount_policy == 'without_discount':
+            # This part is pretty much a copy-paste of the method '_onchange_discount' of
```
Comment (JKE-be) [inline addons/website_sale/models/sale_order.py]: in this case, can we call this method instead to duplicate all this code ?

Comment (nim-odoo) [reply addons/website_sale/models/sale_order.py]: There is always a way to do it.

In this case, note that the SO line might not exist yet:

https://github.com/odoo/odoo/blob/98fe5054c26828a3a15c8cd7b5905c7cf99c6103/addons/website_sale/models/sale_order.py#L169-L171

So we'd also have to refactor the onchange.

**Comment by nim-odoo** [comment]
robodoo r+

**Pull request closed without merging**